### DISCUSSION TITLE
Looking for a Frigate/HA hardware recommendation.

### DISCUSSION THREAD

**Discussion by mcdent** [blakeblackshear/frigate#2449] [Ask A Question]
Hey, I'm looking to setup a new CCTV system to replace my sloooooow Synolgy NVR. 
I have 6 cameras, all Hikvision 4mp, apart from one Annke C800. 
I currently  run a small Homeassistant instance on a Raspberry PI.

I'm wondering what hardware to look at, to replace the Synology and RPI with HA and Frigate. 
I would like to be able to have plenty of storage so I can store recordings, maybe for 14 days history?

I need the ability to access live camera views when I am away from home, that's one area the Synology is very poor, it takes about 20 seconds to begin displaying the camera streams when you open the app.

Any pointers towards something which is not too big, not too much power use and has storage for recordings would be very welcome!
Thanks

**Comment by blakeblackshear** [comment]
There are recommendations in the docs: https://docs.frigate.video/hardware

**Reply by mcdent** [reply]
Thanks for that. Just wondering what 'several 1080p cameras' ? 6 to 10?
I see I can get the Minisforum GK50  in the UK. Looks like this takes a standard 2.5" SSD?
Would you also recommend the USB Coral?

Thanks

**Reply by blakeblackshear** [reply]
Any Coral is fine. I haven't pushed the limits, but a J4125 processor can handle 4 1080p cameras without breaking a sweat. There are a few versions of the GK50. I ended up just using an external USB3 drive for simplicity.

**Reply by mcdent** [reply]
Hey Blake, thanks. I got a GK50 which has a M2 slot but I am using a SSD instead, so don't need the M2 slot for storage.
 In your hardware page you mentioned that the GK50 supports USB Coral. Is there any reason that my GK50 with M2 will not support a Coral M2? I'm finding the USB coral impossible to get in the UK. Thanks

**Reply by blakeblackshear** [reply]
Not all M2 slots support the Coral. Many are only for storage devices. I don't think the GK50 supports it.

**Reply by zoltanszalontay** [reply]
Google makes Coral with different m.2 key form factors, though all of them are either phased out or in 40+wk backlog. https://coral.ai/products/#production-products

Am I right that an A+E key version, instead of a B+M which mini PC is also hard to find btw, would work in a mini PC having an m.2 A+E slot?

**Comment by t0ny-peng** [comment]
Not sure if mine is overkill but it definitely works very well.

Intel NUC(i5-52xx), 16GB memory, 2TB NVME in the M.2 slot + 512GB SSD(via USB3.0) + 16TB HDD(vis USB 3.0). + Coral USB TPU. The bare machine runs Proxmox, and Frigate runs as a docker container. Home assistant runs in a single VM. 

P.S., I found a special USB type A to USB type C **dual-male** dongle which avoids the long cable. 

![image](https://user-images.githubusercontent.com/4816327/146692719-8f0b1122-8572-4be4-ad55-b9eaa30ed4cd.png)
